File: python-app/models/pivot_table/data_source.py
from types import SimpleNamespace

class DataSource(SimpleNamespace):
    def __init__(self, files: list[str], merged_file: str | None) -> None:
        self.files = files
        self.merged_file = merged_file
    
    def to_dict(self) -> dict:
        return {
            "files": self.files,
            "merged_file": self.merged_file
        }

    @classmethod
    def from_json(cls, json_data: dict[str, any]) -> "DataSource":
        return cls(
            files=json_data.get("files", []),
            merged_file=json_data.get("merged_file")
        )

File: lib/models/pivot_table/data_source.dart
class DataSource {
  final List<String> files;
  final String? mergedFile;

  DataSource({required this.files, this.mergedFile});

  Map<String, dynamic> toJson() {
    return {"files": files, "merged_file": mergedFile};
  }

  factory DataSource.fromJson(Map<String, dynamic> json) {
    return DataSource(
      files: List<String>.from(json["files"]),
      mergedFile: json["merged_file"],
    );
  }

  DataSource copyWith({List<String>? files, String? mergedFile}) {
    return DataSource(
      files: files ?? this.files,
      mergedFile: mergedFile ?? this.mergedFile,
    );
  }
}
